lib/WebService/Client.pm

```diff
@@ -103,6 +100,26 @@ sub log {
     $self->logger->$log_method($msg);
 }
 
+sub prepare_response {
+    my ($self, $res, $req, $num_attempts) = @_;
+    Moo::Role->apply_roles_to_object($res, 'HTTP::Response::Stringable');
+    return;
+}
+
+sub parse_response {
```

Comment:
The `req` method parses the response in a rather opinionated fashion:

- 404 / 410 are converted to `undef` (What if you want to inspect the response content?)
- Dies on 1xx / 3xx / 4xx / 5xx
- On 2xx, content is assumed to be JSON

So let's just take two examples:

- WS::Client will die on 2xx, non-JSON responses
    + It'll try to parse them as JSON
    + ... which will fail (e.g. for XML) and die with an error
- You *must* die on non-2xx responses. Even 1xx or 3xx, which might not be expected or desirable.